src/new-client/src/App/Trades/TradesGrid/TradesGrid.tsx

```diff
@@ -145,47 +183,79 @@ const DATE_FORMAT = "dd-MMM-yyyy"
 
 const formatTo6Digits = significantDigitsNumberFormatter(6)
 
+interface RowParam {
+  data: Trade[]
+  index: number
+  style: React.CSSProperties
+}
+
+const VirtualizedTableRow = ({ data, index, style }: RowParam) => (
+  <TableBodyRow key={data[index].tradeId} style={style}>
+    <StatusIndicator status={data[index].status}>
+      0
+      <span className="spacer" />
+    </StatusIndicator>
+    {colConfigs.map(({ field, numeric, valueFormatter, width }) => (
+      <TableBodyCell key={field} numeric={numeric} width={width}>
+        {valueFormatter?.(data[index][field]) ?? data[index][field]}
+      </TableBodyCell>
+    ))}
+  </TableBodyRow>
+)
+
+const convertRemToPixels = (rem: number) => {
+  return rem * parseFloat(getComputedStyle(document.documentElement).fontSize)
+}
+
 export const TradesGrid: React.FC = () => {
   const trades = useTableTrades()
   const tableSort = useTableSort()
+  const [tableHeight, setTableHeight] = useState(55)
+  const containerPercent = useSelectedHeight()
+
+  useEffect(() => {
```

Comment:
If you have to do something like this, I would use  useLayoutEffect.  For the reasons described here: 

https://daveceddia.com/useeffect-vs-uselayouteffect/

In general, you are doing all of this to get one value, the dynamic table height.  What I would I do, in that case, is define one custom hook, `useTableHeight`, composed of Observable streams that emits the right table-height.  Also, where are the number values that you are passing into your functions--`0.01`, `3.5`, `5`, `2`, etc.--coming from?

At the end of the day, though, the question is do we need to do all this work?  What were the approaches css approaches you tried before dynamically calculating the height with JS?

Reply:
Yeah as I said, this version works but not that elegant, since those values like 0.01, 3.5  are from rem values of other element. I do want to use CSS way to use all the available space in that area, but the react-window library needs to use absolute styling. Still thinking about a better way, can we try to use some other library to virtualize the grid?